#162573546 creates a user class

ebenezermakinde opened the pull request:
#### What does this PR do?

* Implement user class

#### Description of Task to be completed?

* create a user class
* add jsdoc
* add user route
* add helpers/helper
* add user route

#### How should this be manually tested?

* Using POSTMAN

#### Any background context you want to provide?

* Not applicable

#### What are the relevant pivotal tracker stories?

* A user should be able to register

#### Screenshots (if appropriate)

* Not applicable

#### Questions:

* Not applicable

Commit:
feat(user-class): creates a user class

- creates a user class
- adds jsdoc
- adds user route
- adds helpers/helper
- adds user route

[Finishes #162573546]

otseobande commented on server/helpers/helper.js:
```diff
@@ -1,9 +1,97 @@
+   * @param {object} res
+   * @param {object} query
+   * @param {object} values
+   * @returns {object} a message
```
The return type here is a `Promise` since `db.query` is a promise.

otseobande commented on server/models/ireporterModel.js:
```diff
@@ -0,0 +1,29 @@
+   * @param {Array} params
+   * @returns {object} promise
+   */
+  query(text, params) {
```
I don't see the value this adds to just using `pool.query` directly.

ebenezermakinde replied:
Please can you shed more light on this feedback, I don't really understand it

ebenezermakinde replied:
I was actually using a lot of queries here. But since I started using this it actually helped me reduce my lines of code and helped me with a lot more manipulations. I actually saw this at codementor, and I realized it's usefulness

otseobande commented on server/routes/UserRoute.js:
```diff
@@ -0,0 +1,10 @@
+// router.post('/auth/login', user.canLogin);
+router.post('/auth/signup', user.canSignup);
+// router.patch('/red-flags/:id/status', user.canEdit);
+// router.patch('/interventions/:id/status', user.canEdit);
```
Please don't leave commented lines in the codebase

ebenezermakinde replied:
Yes. I will remove it.

otseobande commented on server/controller/IncidentController.js:
```diff
@@ -1,256 +1,133 @@
+   * Get all incidents
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} all incident objects
```
If a function does not use the keyword `return` in the body it returns `undefined`.

ebenezermakinde replied:
Seen. I am making the changes.

otseobande reviewed (commented):
Please address these comments.

Commit:
feat(create user): create a user

- include return for all helper function
- remove commented code in userroute
- refactor forloop to foreach

[Finishes #162573546]
